### ISSUE TITLE
Multivariate Multihead New Model Architecture

### ISSUE THREAD

**Issue by farzanmrz** [farzanmrz/spread-res#44]
This issue tracks the complete re-architecture of the model based on the transition to a multi-head, OHE-metadata-driven, cross-attribute prediction design.

### Key changes included:
- [ ] Switch from `BCEWithLogitsLoss` to `CrossEntropyLoss` with multi-class outputs. This we might test with existing approach and see if it works before proceeding
- [ ] Redefine spreadsheet representation using OHE for all metadata
- [ ] Building on above point this relates to back excel parsing pipeline and the checking for is_capital that needs to be done moreover the font_size encoding that needs to be done. So this goes from the back
- [ ] Decide if `x_metadata` and `y_tok` should be split vs. fused. Since y_tok is essentially the metadata and so is x_metadata but till now it was all metadata except bold/non-bold due to it being target variable and then font_size due to above issue mentioned which we will try to tackle with encoding it now
- [ ] Fundamentally the forward method and training method since it triggers forward will both change and they overlap so lets go by model first
- [ ] Design separate heads in `forward` pass for each metadata type at the end which you call binary_classifier, it should return out logits values as many num of classes exist per head for whatever metadata
- [ ] Create dynamic masking matrix (`x_metask`) during training who's job is masking metadata that is gonna be predicted by an individual head and this might also not be passed into forward this might exist internally since we have fixed locations of masking out stuff.
- [ ] Ensure metadata masking happens at runtime, not precomputed and stored as matrix in DS object. Therefore we make masking like a index location along third dim apparently that is what works
- [ ] Adjust model outputs per head (e.g., 2 logits for bold, 3 for underline)
- [ ] Update training and evaluation pipeline accordingly. BCEWithLogitsLoss turns to CELoss, I know first step is trying current approach but with CELoss this has to happen first but since logically training happens here we put this at top and also here
- [ ] In inference also there is some adjustment for two differing model types with metadata and without and same with evaluation so firstly that needs to be looked at.
- [ ] More importantly the very approach of inferencing has got to change alsong with how we are evaluating it since all metadata is being predicted now.

### Notes:
- This is an umbrella issue and will be broken into child issues.
- Subtasks that fall into current sprint will be labeled and assigned as needed.
